src/mini_allocator.c

```diff
@@ -690,6 +744,35 @@ mini_deinit(cache *cc, uint64 meta_head, page_type type, bool pinned)
    } while (meta_addr != 0);
 }
 
+/*
+ *-----------------------------------------------------------------------------
+ * mini_deinit -- De-Initialize a new mini allocator.
+ *
+ *      This is the last thing to do to release all resources acquired by the
+ *      mini-allocator. All pages & extents reserved or allocated by the mini-
+ *      allocator will be released. The mini-allocator cannot be used after this
+ *      call, and will need to go through mini_init().
+ *
+ * Returns: Nothing.
+ *-----------------------------------------------------------------------------
+ */
+void
+mini_deinit(mini_allocator *mini, key start_key, key end_key)
```

Comment:
I have tried to consolidate different things into this one single `deinit` method, which will be symmetric to `mini_init()`.

I think I got it right ... but the addition of this interface merits discussion.

Also other interfaces such as `mini_release()`, `mini_destroy_unused()` ... are somehow 'overlapping' in their names and semantics.

It will be a good opportunity as part of this rework to review those interfaces and normalize their names / functioning.